### PR TITLE
feat: add String extension for URL encoding in UTF-8 and UTF-16 format

### PR DESCRIPTION
**Pull request by vincentdelmundo**
# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data **before every commit**, including API endpoints and keys
- [x] I checked that all possible internal and platform/library exceptions are gracefully handled and **will not** crash the host app
- [x] I checked for open & known issues (especially crashes) for any newly added platform APIs or libraries
- [x] I tested non-trivial changes on a real device
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

File: sdk-utils/src/main/kotlin/com/rakuten/tech/mobile/sdkutils/StringExtension.kt [hunk 1]
```diff
@@ -106,4 +108,22 @@ object StringExtension {
...
         }
     }
+
+    /**
+     * Get the URL encoded value of the string for charset "UTF-8".
+     *
+     * @return UTF-8 URL encoded value
+     */
+    fun String.getUTF8UrlEncoded(): String {
+        return URLEncoder.encode(this, StandardCharsets.UTF_8.displayName())
```
Comment (leila-gu) [inline sdk-utils/src/main/kotlin/com/rakuten/tech/mobile/sdkutils/StringExtension.kt]: It seems that `URLEncoder.encode` can throw `UnsupportedEncodingException`, maybe better to wrap it in try-catch block?

Comment (vincentdelmundo) [reply sdk-utils/src/main/kotlin/com/rakuten/tech/mobile/sdkutils/StringExtension.kt]: updated

**Review by leila-gu — APPROVED** [review]
LGTM

**Review by maureenorea-clores — APPROVED** [review]
LGTM

**Review by mdusmangani-jp — APPROVED** [review]
LGTM

**Pull request merged**